extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -85,11 +85,19 @@ export class ProjectsController {
 				// TODO: prompt to create new datasources.json; for now, swallow
 			}
 			else {
+				this.projects = this.projects.filter((e) => { return e !== newProject; });
 				throw err;
 			}
 		}
 
-		this.refreshProjectsTree();
+		try {
+			this.refreshProjectsTree();
+		}
+		catch (err) {
+			// if the project didnt load - remove it from the list of open projects
+			this.projects = this.projects.filter((e) => { return e !== newProject; });
```

Comment:
I think VS/SSDT shows the project in the tree, but with a "failed to load" message.  I like the idea of doing that here, too, but maybe that's worth doing in a separate task.

Reply:
yeah we can do that with a reload context menu button. Let's go with this for now t unblock and do that as a seperate thing.

Reply:
Sounds good!  Created as #11047